File: 02-programacion-basica/06-funciones-y-metodos/java/soluciones/RetoCalculadora.java
public class RetoCalculadora {
    public static void main(String[] args) {
        System.out.println(sumar(8, 2));
        System.out.println(restar(8, 2));
        System.out.println(multiplicar(8, 2));
        System.out.println(dividir(8, 2));
    }

    static double sumar(double a, double b) { return a + b; }
    static double restar(double a, double b) { return a - b; }
    static double multiplicar(double a, double b) { return a * b; }

    static double dividir(double a, double b) {
        // Dividir entre cero no produce un resultado válido para este problema.
        if (b == 0) throw new IllegalArgumentException("Divisor igual a cero");
        return a / b;
    }
}

File: 02-programacion-basica/06-funciones-y-metodos/scala/soluciones/RetoCalculadora.scala
def sumarReto(a: Double, b: Double): Double = a + b
def restarReto(a: Double, b: Double): Double = a - b
def multiplicarReto(a: Double, b: Double): Double = a * b
def dividirReto(a: Double, b: Double): Double =
  // require detiene la función si el divisor no cumple la precondición.
  require(b != 0, "Divisor igual a cero")
  a / b

@main def retoCalculadora(): Unit =
  println(sumarReto(8, 2))
  println(restarReto(8, 2))
  println(multiplicarReto(8, 2))
  println(dividirReto(8, 2))
